### PR TITLE
[ENHANCEMENT] Create ImportTransactionsUseCase — move CSV parsing out of ImportDbViewModel (issue #7)

### LINKED ISSUE #7: [ARCH] Create ImportTransactionsUseCase — move CSV parsing out of ImportDbViewModel
## Problem
`ImportDbViewModel` contains CSV parsing logic and calls `repository.createTransaction()` directly. Both concerns belong in the domain layer, not the ViewModel.

```kotlin
// ImportDbViewModel — business logic and direct repo access
repository.createTransaction(parsedTransaction)
```

## Fix
1. Create `ImportTransactionsUseCase` in `domain/usecase/transaction/` that owns the CSV parsing and delegates transaction creation to `CreateTransactionUseCase`.
2. `ImportDbViewModel` should only call the new use case and handle UI state.
3. Register in `useCaseModule.kt`.

## Files
- `presentation/ui/db/ImportDbViewModel.kt`
- `di/module/viewModelModule.kt`

### PR DESCRIPTION
**Pull request by Benoist-CID**
## Summary

- Create `ImportTransactionsUseCase` in `domain/usecase/transaction/` that owns CSV line splitting, parsing via `transactionFromCsv()`, and delegation to `CreateTransactionUseCase`; malformed lines are silently skipped (preserving original behaviour)
- Refactor `ImportDbViewModel`: remove `FinancialRepository`, inject `ImportTransactionsUseCase`; `importDb()` becomes a single use case call
- Register `ImportTransactionsUseCase` in `useCaseModule` via `factoryOf`
- Update `viewModelModule` to inject the new use case into `ImportDbViewModel`

Closes #7

## Test plan

- [x] `ImportTransactionsUseCaseTest` — happy path: all valid lines parsed and persisted
- [x] `ImportTransactionsUseCaseTest` — edge case: malformed lines skipped, valid ones still imported
- [x] `ImportTransactionsUseCaseTest` — edge case: empty string produces zero calls to `CreateTransactionUseCase`
- [x] `./gradlew assembleDebug` — BUILD SUCCESSFUL
- [x] `./gradlew test` — all tests passing

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: app/src/main/java/fr/laforge/benoist/financialmanager/domain/usecase/transaction/ImportTransactionsUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+     * @param csv Raw CSV string where each line represents one transaction.
+     * @return The number of transactions successfully imported.
+     *
+     * @note Malformed lines are caught and logged at warning level, then skipped.
+     * The caller receives the count of successfully persisted rows so it can surface
+     * a meaningful result to the user if needed.
+     */
+    suspend operator fun invoke(csv: String): Int {
+        var imported = 0
+        csv.split('\n').forEach { line ->
```
Comment (Benoist-CID) [inline app/src/main/java/fr/laforge/benoist/financialmanager/domain/usecase/transaction/ImportTransactionsUseCase.kt]: You should create a constant in the companion object for the separator. IT is cleaner

File: app/src/main/java/fr/laforge/benoist/financialmanager/presentation/ui/db/ImportDbViewModel.kt [hunk 2]
```diff
@@ -5,38 +5,51 @@ import androidx.compose.runtime.mutableStateOf
...
 
+    /**
+     * Triggers the import of all transactions encoded in [toBeImported].
+     *
+     * Delegates entirely to [ImportTransactionsUseCase]; malformed lines are
+     * silently skipped by the use case.
+     */
     fun importDb() {
         viewModelScope.launch {
             withContext(Dispatchers.IO) {
```
Comment (Benoist-CID) [inline app/src/main/java/fr/laforge/benoist/financialmanager/presentation/ui/db/ImportDbViewModel.kt]: We shoudl inject the dispatcher

File: app/src/main/java/fr/laforge/benoist/financialmanager/domain/usecase/transaction/ImportTransactionsUseCase.kt (hunk 1 above)
Comment (Benoist-CID) [reply app/src/main/java/fr/laforge/benoist/financialmanager/domain/usecase/transaction/ImportTransactionsUseCase.kt]: Done — extracted to \ and updated the \ call to use it.

File: app/src/main/java/fr/laforge/benoist/financialmanager/presentation/ui/db/ImportDbViewModel.kt (hunk 2 above)
Comment (Benoist-CID) [reply app/src/main/java/fr/laforge/benoist/financialmanager/presentation/ui/db/ImportDbViewModel.kt]: Done — added \ to the constructor and replaced the hardcoded \ in \ with it. The default keeps production behaviour unchanged while allowing tests to override.
